### DISCUSSION TITLE
Feature: run task *only* from cache

### DISCUSSION THREAD

**Discussion by tnorrisvnc** [vercel/turborepo#7474] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

My CI pipeline writes some metadata files, including the build number, when deploying a package. These metadata files are cached as outputs of a "deploy" task in turborepo. 

When a subsequent pipeline runs, I'd like to be able to do an up-front check: 
(1) check if the current code needs to be published (ie., is not cached)
(2) pull down, open and print information from the cached metadata files, if any

But I don't want this step of the build to _actually run_ the "deploy" task in case of a cache miss.

### Describe the solution you'd like

Something like a `--cached-only` flag that either skips uncached tasks or errors out when one is found.

### Describe alternatives you've considered

* Call turbo with `--dry-run` first, check, and then call turbo again without `--dry-run` only if the task is cached. 
  * This works, but it's a little awkward and I don't like the risk that the 2nd call might actually run 'deploy' if anything happens to the cache, particularly if it's a remote cache
* Use an environment variable to skip actually running tasks
  * I haven't actually tried this; considered it as a failsafe but I'd prefer not to make tasks more sensitive to environment variables

**Comment by mehulkar** [comment]
I'm not opposed to this as a feature, it seems like a good win to me. I'd be interested in ironing out an API that can pick and choose any combination of caches (fs or remote) and execution (real or dry).
